generator/sponsored_dataset.cpp

```diff
@@ -0,0 +1,213 @@
+#include "generator/sponsored_dataset.hpp"
+
+#include "platform/local_country_file.hpp"
+#include "platform/local_country_file_utils.hpp"
+#include "platform/platform.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include "std/fstream.hpp"
+#include "std/iostream.hpp"
+#include "std/limits.hpp"
+
+namespace generator
+{
+namespace
+{
+string EscapeTabs(string const & str)
+{
+  stringstream ss;
+  for (char c : str)
+  {
+    if (c == '\t')
+      ss << "\\t";
+    else
+      ss << c;
+  }
+  return ss.str();
+}
+}  // namespace
+
+SponsoredDataset::AddressMatcher::AddressMatcher()
```

Comment:
Could you please add separating comments for groups of class methods definitions, like:
```c++
// SponsoredDataset::AddressMatcher M-u 64 -
```